numba/core/typing/arraydecl.py

```diff
@@ -415,6 +415,11 @@ def resolve_itemset(self, ary, args, kws):
     def resolve_nonzero(self, ary, args, kws):
         assert not args
         assert not kws
+        if ary.ndim == 0 and numpy_version >= (2, 1):
```

Comment:
I think technically this should be an error raised at runtime, but catching it early during compilation is probably more useful for users.